internal/dev/kubernetes_cluster.go

```diff
@@ -0,0 +1,307 @@
+package dev
+
+import (
+	"bytes"
+	"context"
+	goerrors "errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"sort"
+	"time"
+
+	"github.com/go-logr/logr"
+	configv1 "github.com/openshift/api/config/v1"
+	operatorsv1 "github.com/operator-framework/api/pkg/operators/v1"
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+	"sigs.k8s.io/yaml"
+
+	aoapis "github.com/openshift/addon-operator/apis"
+)
+
+// Container object to hold kubernetes client interfaces and configuration.
+type KubernetesCluster struct {
+	Scheme     *runtime.Scheme
+	Config     *rest.Config
```

Comment:
Is this only needed for informational purpose around the respective cluster in the future? Because I guess any functional purpose would be served `CtrlClient client.Client`

Reply:
The idea is that `KubernetesCluster` is grouping multiple different clients, controller-runtime, core, dynamic, rest mapper...
There are plenty clients to go around :)

If you don't have the client that you need, having the config exposed makes it easier to create the one that you need for testing.

Reply:
Gotcha! Makes sense :)